ScientificCalculator/src/ParserExpr.hs

```diff
@@ -0,0 +1,173 @@
+module ParserExpr where
+
+import Text.Parsec
+import Text.Parsec.String (Parser)
+import Text.Parsec.Expr
+import qualified Text.Parsec.Token as Token
+import Text.Parsec.Language (emptyDef)
+import Control.Monad.Identity
+
+import Expr
+
```

Comment:
Please, kindly read the readme I explained there how the calculator works, with some examples, I also included help command in the calculator. 
For expression evaluation you need to use first (=), then type the expression to evaluate. What you are trying to do is, first apply "sin" to the last result and then entering a constant, basically this:
sin (lastResult) π.
Which is not a valid expression, because there is no binary operation joining both expressions and triggers the error.
The (m+) works as expected for my calculator: Adds the last computed result to the value stored in memory.
If you want to add something new, first:
Evaluate the value that you want to add, little example:

> 45
Result: 45.0
> m+
> mr
Memory: 45.0
> =
Enter an expression to evaluate:
sin π
Result: 1.2246467991473532e-16
> mc
Memory cleared
> m+
> mr
Memory: 1.2246467991473532e-16
> 

the calculator is based on the one on macOS which has a similar functionality

Reply:
Gotcha, you have another readme.